reproducibility_project/src/engines/mcccs/conv_traj.py

```diff
@@ -0,0 +1,113 @@
+"""Reads the .xyz files for different seeds in NPT MC simulations, and combines and saves them in the hdf5 format."""
+import os
+import shutil
+from glob import glob
+
+import freud
+import matplotlib.pyplot as plt
+import mdtraj as md
+import numpy as np
+import signac
+from scipy import stats
+
+
+def main():
+    """Read xyz files from each replica and combine them all into one big trajectory."""
+    data_path = "analysis_data"
+    # if os.path.exists(data_path):
+    #    shutil.rmtree(data_path)
+    # os.makedirs(data_path)
+    # delete the folder manually
+    if not os.path.isdir(data_path):
+
+        os.makedirs(data_path)
+
+    os.chdir(data_path)
+
+    project = signac.get_project()
+
+    for (molecule, ensemble, temperature, pressure), group in project.groupby(
+        (
+            "molecule",
+            "ensemble",
+            "temperature",
+            "pressure",
+        )
+    ):
+        print(molecule, ensemble, temperature, pressure)
+        if not os.path.isdir(
+            "{}_{}_{}K_{}kPa".format(molecule, ensemble, temperature, pressure)
+        ):
+
+            os.makedirs(
+                "{}_{}_{}K_{}kPa".format(
+                    molecule, ensemble, temperature, pressure
+                )
+            )
+        os.chdir(
+            "{}_{}_{}K_{}kPa".format(molecule, ensemble, temperature, pressure)
+        )
+        base_dir = os.getcwd()
+        if ensemble == "NPT" and molecule == "methaneUA":
+            if os.path.exists(os.path.join(os.getcwd(), "comb_traj.h5")):
+                continue
+            traj_list = []
+            for job in group:
+                # print(job)
+                os.chdir(job.ws)
+                traj_files = sorted(glob("box1movie1a*prod*pdb*"))
+                if len(traj_files) < 4:
+                    # print("warning, only {}  prod cycles complete for {} {} {} {} {}".format(len(traj_files),job, molecule, ensemble, temperature, pressure))
+                    print(
+                        "warning, only {}  prod cycles complete for {} {} {} {} {}. So not inlcuding its xyz files.".format(
+                            len(traj_files),
+                            job,
+                            molecule,
+                            ensemble,
+                            temperature,
+                            pressure,
+                        )
+                    )
+                    continue
+                prod_number = 0
+                for filename in traj_files:
+                    print(
+                        "The filename is {} and the prod number is {}. These two should match.".format(
+                            filename, prod_number
+                        )
+                    )
+                    traj = md.load(filename, top="init1.mol2")
```

Comment:
Are you confident that loading the entire trajectory into memory at once wouldn't take too much memory?

Reply:
https://www.mdtraj.org/1.9.5/api/generated/mdtraj.iterload.html?highlight=iterload#mdtraj-iterload

This will let you chunk your traj if you need specific data along the way.

Reply:
It was working for me with 2 gb memory (but I only tested methane). I will look into iterload if memory becomes an issue.